src/User.ts

```diff
@@ -124,4 +125,8 @@ export class User {
         Logger.createStatic("User", "fromStorageString");
         return new User(JSON.parse(storageString));
     }
+
+    public async dpopProof(url: string, httpMethod?: string): Promise<string> {
```

Comment:
It feels a bit weird exposing this here: there needs to be a way of allowing consumers to generate a DPoP proof and then attaching that to authenticated requests.

But it seemed like the easiest way from the pov of the react-oidc-client library... for example https://github.com/dbfr3qs/react-oidc-context/blob/217d6d18109fc82526d7abc4015a98ec359fe79b/example/index.tsx#L29C13-L39C16